server/test/unit/server/webservices/views/test_content.py

```diff
@@ -0,0 +1,54 @@
+import unittest
+
+import mock
+
+from .base import assert_auth_UPDATE
+from pulp.server.exceptions import InvalidValue
+from pulp.server.webservices.views.content import UploadSegmentResourceView
+
+
+class TestUploadSegmentResourceView(unittest.TestCase):
+    """
+    Tests for views for uploads to a specific id.
+    """
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_UPDATE())
+    @mock.patch('pulp.server.webservices.views.content.generate_json_response')
+    @mock.patch('pulp.server.webservices.views.content.factory')
+    def test_put_upload_segment_resource(self, mock_factory, mock_serializer):
+        """
+        Test the UploadSegmentResourceView under normal conditions
+        """
+        mock_upload_manager = mock.MagicMock()
+        mock_factory.content_upload_manager.return_value = mock_upload_manager
+        request = mock.MagicMock()
+        request.body = 'upload these bits'
+
+        upload_segment_resource = UploadSegmentResourceView()
+        response = upload_segment_resource.put(request, 'mock_id', 4)
+
+        mock_upload_manager.save_data.assert_called_once_with('mock_id', 4, 'upload these bits')
+        mock_serializer.assert_called_once_with(None)
+        self.assertTrue(response is mock_serializer.return_value)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_UPDATE())
+    @mock.patch('pulp.server.webservices.views.content.factory')
+    def test_put_upload_segment_resource_bad_offset(self, mock_factory):
+        """
+        Test the UploadSegmentResourceView with an invalid offset value (not an int)
+        """
+        mock_upload_manager = mock.MagicMock()
+        mock_factory.content_upload_manager.return_value = mock_upload_manager
+        request = mock.MagicMock()
+
+        upload_segment_resource = UploadSegmentResourceView()
+
+        try:
+            upload_segment_resource.put(request, 'mock_id', 'invalid_offset')
+        except InvalidValue:
+            pass
```

Comment:
I agree about assertRaises.

I don't think I can really test that a MissingResource exception is raised since that is actually a side effect of the manager; this code doesn't interact with it at all. I think I should just remove the `raises MissingResource` statement from the comments. What do you think @dkliban?